_posts/2022-08-29-sbt-tutorial.md

```diff
@@ -0,0 +1,514 @@
+We can specify the sbt version for the project by using a file called as _build.properties_ under _project_ directory. We can mention the sbt version in the file as:
+```
+sbt.version=1.6.1
+```
+This will ensure that the project will use the sbt version as _1.6.1_ even though your might have installed another version of sbt in your machine.
+
+### Small Scala File and SBT Commands
+Now, let's try to add a simple Scala file with main method. We will keep this file under the directory _src/main/scala_ with package as _com.rockthejvm_
+
+```
+package com.rockthejvm
+object Main {
+  def main(args: Array[String]): Unit = {
+    println("Hello, World!")
+  }
+}
+```
+Let's start the sbt console by executing the command _sbt_ at the root of the project.
+To compile the project, we can use _compile_ in the command. This will compile all the scala files in the project and creates .class files. These files are kept under the directory _target_. 
+** SBT uses a feature called as _Incremented Compilation_  to speed up the compilation. In this way, sbt will compile only the modified files, and re-using the previously compile class files for the unchanged files. **
+We can use _clean_ command to remove all the generated class files.
+
+We can run the application by using the command _run_. If we have multiple main classes in the project, sbt will show a prompt to select the main class to run. 
+In such case, we can use _runMain_ command by passing the class to run as:
+```
+runMain com.rockthejvm.Main 
+```
+
+SBT also allows to do automatic compilation on any source file change. For that, we can start the compile command prefixed by ~ symbol within the sbt shell:
+```
+~compile
+```
+Now, sbt will track the source files and if any change is detected in the files, it will automatically recompile them.
+
+## Adding external Dependenncies
+So far, we have created scala files without using any external dependencies. Now, let's look at how we can add additional libraries to our project. We can add the dependencies to out project by adding it to the settings _libraryDependencies_. SBT follows ivy style format to define the dependencies as :
+```
+libraryDependencies += "com.lihaoyi" %% "fansi" % "0.4.0"
+```
+
+The method `+=` appends the provided library to the project dependencies. 
+An sbt dependency contains mainly 3 parts separated by % symbol. 
+The first part is the groupId of the library. The second part is the library name(artifactId) and the third part is the version of the library to be used. If you notice, you can see that a double percentage symbol (%%) is used between groupId and artifactId. 
+Scala is not binary compatible with different versions (such as 2.11, 2.12, 2.13 etc). Hence there are separate releases for each scala libraries for each required versions. %% symbol ensures that sbt uses the same version of library as the project.
```

Comment:
with -> between
same version -> same Scala version

Reply:
I am already using `between` here. Could you check if that is what you meant?